tests/regression/test_point_eval_api.py

```diff
@@ -0,0 +1,138 @@
+
+    # multiple points as tuple
+    assert np.allclose([[0.2], [0.3]], f.at((0.2, 0.3)))
+    assert np.allclose([[0.2], [0.3]], f.at(((0.2,), (0.3,))))
+    assert np.allclose([[0.2], [0.3]], f.at(([0.2], [0.3])))
+    assert np.allclose([[0.2], [0.3]], f.at((np.array(0.2), np.array(0.3))))
+    assert np.allclose([[0.2], [0.3]], f.at((np.array([0.2]), np.array([0.3]))))
+
+    # multiple points as list
+    assert np.allclose([[0.2], [0.3]], f.at([0.2, 0.3]))
+    assert np.allclose([[0.2], [0.3]], f.at([(0.2,), (0.3,)]))
+    assert np.allclose([[0.2], [0.3]], f.at([[0.2], [0.3]]))
+    assert np.allclose([[0.2], [0.3]], f.at([np.array(0.2), np.array(0.3)]))
+    assert np.allclose([[0.2], [0.3]], f.at([np.array([0.2]), np.array([0.3])]))
+
+    # multiple points as numpy array
+    assert np.allclose([[0.2], [0.3]], f.at(np.array([0.2, 0.3])))
+    assert np.allclose([[0.2], [0.3]], f.at(np.array([[0.2], [0.3]])))
+
+
+def test_2d_args():
+    mesh = UnitSquareMesh(1, 1)
+    f = mesh.coordinates
+
+    # one point
+    assert np.allclose([0.2, 0.4], f.at(0.2, 0.4))
+    assert np.allclose([0.2, 0.4], f.at((0.2, 0.4)))
+    assert np.allclose([0.2, 0.4], f.at([0.2, 0.4]))
+    assert np.allclose([0.2, 0.4], f.at(np.array([0.2, 0.4])))
+
+    # multiple points as arguments
+    assert np.allclose([[0.2, 0.4], [0.3, 0.5]], f.at((0.2, 0.4), (0.3, 0.5)))
+    assert np.allclose([[0.2, 0.4], [0.3, 0.5]], f.at([0.2, 0.4], [0.3, 0.5]))
+    assert np.allclose([[0.2, 0.4], [0.3, 0.5]], f.at(np.array([0.2, 0.4]),
+                                                      np.array([0.3, 0.5])))
+
+    # multiple points as tuple
+    assert np.allclose([[0.2, 0.4], [0.3, 0.5]], f.at(((0.2, 0.4), (0.3, 0.5))))
+    assert np.allclose([[0.2, 0.4], [0.3, 0.5]], f.at(([0.2, 0.4], [0.3, 0.5])))
+    assert np.allclose([[0.2, 0.4], [0.3, 0.5]], f.at((np.array([0.2, 0.4]),
+                                                       np.array([0.3, 0.5]))))
+
+    # multiple points as list
+    assert np.allclose([[0.2, 0.4], [0.3, 0.5]], f.at([(0.2, 0.4), (0.3, 0.5)]))
+    assert np.allclose([[0.2, 0.4], [0.3, 0.5]], f.at([[0.2, 0.4], [0.3, 0.5]]))
+    assert np.allclose([[0.2, 0.4], [0.3, 0.5]], f.at([np.array([0.2, 0.4]),
+                                                       np.array([0.3, 0.5])]))
+
+    # multiple points as numpy array
+    assert np.allclose([[0.2, 0.4], [0.3, 0.5]], f.at(np.array([[0.2, 0.4],
+                                                                [0.3, 0.5]])))
+
+
+def test_dont_raise():
+    mesh = UnitIntervalMesh(1)
+    V = FunctionSpace(mesh, "CG", 1)
+    f = Function(V).interpolate(Expression("2.0 * x[0]"))
+
+    # raise exception without dont_raise
+    with pytest.raises(PointNotInDomainError):
+        f.at(-1)
+
+    # dont_raise=True
+    assert 'None' == str(f.at(-1, dont_raise=True))
+    assert '[None, array(0.0), array(1.0)]' == str(f.at([-1, 0, 0.5], dont_raise=True))
```

Comment:
These feel somewhat brittle, what happens if the string representation of something else turns out to be None?